app/controllers/accounts_controller.rb

```diff
@@ -23,14 +23,15 @@ def create
 
   private
 
-  helper_method memoize def account
-    Account.visible_for(current_user).find(ps.fetch(:id))
+  memoize def public_shared_accounts
+    Set.new(AccountShare.accepted.for_public.pluck(:account_id))
   end
 
-  helper_method memoize def read_only?(id)
-    return false if current_user.account.id == account.parent.id ||
-                    AccountShare.accepted.for(current_user).pluck(:account_id).include?(id)
+  helper_method memoize def public_share?(account)
```

Comment:
```suggestion
  helper_method def public_share?(account)
```

no need to memoize it as a value is search very fast in set